dashboard/app/controllers/courses_controller.rb

```diff
@@ -88,17 +88,17 @@ def create
   end
 
   def update
-    course = Course.find_by_name!(params[:course_name])
+    course = UnitGroup.find_by_name!(params[:course_name])
     course.persist_strings_and_scripts_changes(params[:scripts], params[:alternate_scripts], i18n_params)
     course.update_teacher_resources(params[:resourceTypes], params[:resourceLinks])
     # Convert checkbox values from a string ("on") to a boolean.
     [:has_verified_resources, :visible, :is_stable].each {|key| params[key] = !!params[key]}
     course.update(course_params)
-    redirect_to course
+    redirect_to course_path(course)
```

Comment:
if not specified, `redirect_to` looks for url path helpers based on the type of the object you pass it, in this case looking for the non-existent `unit_group_path` helper.

Reply:
So when you rename the table will this no longer be needed?

Reply:
same as above, this could be fixed by changing the `resources :courses` declaration in routes.rb.